### PR TITLE
Add constraint validation support

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/3054
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests
- [ ] Updated the spec

### CODE REVIEW

File: ballerina/tests/consume_constraint_tests.bal [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+    check produceMessage(child.toString(), CONSTRAINT_VALID_RECORD_QUEUE);
+    Client 'client = check new (DEFAULT_HOST, DEFAULT_PORT);
+    Child|error result = 'client->consumePayload(CONSTRAINT_VALID_RECORD_QUEUE);
+    if result is error {
+        test:assertFail(result.message());
+    } else {
+        test:assertEquals(result, child);
+    }
+    check 'client->close();
+}
```
Comment (chamil321) [inline ballerina/tests/consume_constraint_tests.bal:322]: Do we have any test case by disabling the config?

Comment (dilanSachi) [reply ballerina/tests/consume_constraint_tests.bal:322]: Added in https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/470/commits/7342c94795ad8287bc499803c68ecd8e27849a0b

File: native/src/main/java/io/ballerina/stdlib/rabbitmq/RabbitMQUtils.java [hunk 2]
```diff
@@ -216,6 +230,24 @@ private static Type getPayloadType(Type definedType) {
         return definedType;
     }
 
+    public static Object validateConstraints(Object value, BTypedesc bTypedesc, boolean constraintValidation) {
```
Comment (Bhashinee) [inline native/src/main/java/io/ballerina/stdlib/rabbitmq/RabbitMQUtils.java:233]: This return value is not used anywhere. Do we need to return it?

File: native/src/main/java/io/ballerina/stdlib/rabbitmq/MessageDispatcher.java [hunk 3]
```diff
@@ -188,8 +199,11 @@ private Object[] getResourceParameters(byte[] message, Envelope envelope, AMQP.B
                             returnErrorValue("Invalid remote function signature");
                         }
                         messageExists = true;
-                        arguments[index++] = createAndPopulateMessageRecord(message, envelope,
+                        Object record = createAndPopulateMessageRecord(message, envelope,
                                 properties, referredType);
+                        validateConstraints(record, getElementTypeDescFromArrayTypeDesc(ValueCreator
+                                .createTypedescValue(parameter.type)), constraintValidation);
+                        arguments[index++] = record;
                         arguments[index++] = true;
```
Comment (Bhashinee) [inline native/src/main/java/io/ballerina/stdlib/rabbitmq/MessageDispatcher.java:207]: As a suggestion, we can move this `arguments[index++] = true;` line to the end of the switch and do it once.

**Review by Bhashinee — APPROVED** [review]

**Review by chamil321 — APPROVED** [review]

**Pull request merged**